client/src/components/Category/index.js

```diff
@@ -170,91 +144,132 @@ class Category extends Component {
-                      <ResultCard
-                        extraClass="result-cards-caregory"
-                        idChirty={idChirty}
-                        key0={id}
-                        isActive={isActive}
-                        onClickCompare={() =>
-                          this.changeActive(item.id, item.idChirty)
+                  <CountContext.Consumer>
+                    {consumer => {
+                      const { charityList, changeActive } = consumer;
+                      const charityListId = charityList.map(
+                        charity => charity.charityId
+                      );
+                      const dataActive = data.map(charity => {
+                        if (charityListId.includes(charity.idChirty)) {
+                          charity.isActive = true;
+                        } else {
+                          charity.isActive = false;
                         }
-                        onClickDonate={() => this.handlerPageDonate()}
-                        logo={logo}
-                        name={this.capitalFirst(name)}
-                        text={this.capitalFirst(this.stringIsMore(name, text))}
-                        financial={this.handleFromat(
-                          'financial',
-                          financial,
-                          '#DIV/0!',
-                          6
-                        )}
-                        governance={this.handleFromat(
-                          'governance',
+                        return charity;
+                      });
+                      return dataActive.slice(0, 5).map(item => {
+                        const {
+                          idChirty,
+                          id,
+                          isActive,
+                          logo,
+                          name,
+                          text,
                           governance,
-                          ' -   ',
-                          8
-                        )}
-                        impact={this.handleFromat('impact', impact, ' -   ', 3)}
-                      />
-                    );
-                  })}
+                          impact,
+                          financial,
+                        } = item;
+                        return (
+                          <ResultCard
+                            extraClass="result-cards-caregory"
+                            idChirty={idChirty}
+                            key0={id}
+                            isActive={isActive}
+                            onClickCompare={() =>
+                              changeActive(item.idChirty, item.name)
+                            }
+                            onClickDonate={() => this.handlerPageDonate()}
+                            logo={logo}
+                            name={this.capitalFirst(name)}
+                            text={this.capitalFirst(
+                              this.stringIsMore(name, text)
```

Comment:
@asmaa-ezz (Maybe for the next project) there are some css properties that do somewhat similar things as these capitalFirst and stringIsMore functions, if you want to check them out:
the `text-transform` property lets you capitalize text or convert it to all caps ([link](https://developer.mozilla.org/en-US/docs/Web/CSS/text-transform))

and the `text-overflow` property will automatically truncate text and add "..." if it overflows its container, though this one can be tricky to get working sometimes (you also need `overflow: hidden; white-space: nowrap;` on the element and it has to be text that stays on a single line, not flows to multiple lines). ([link](https://developer.mozilla.org/en-US/docs/Web/CSS/text-overflow))